src/library/psi_library.jl

```diff
@@ -1149,3 +1149,415 @@ function build_two_zone_5_bus(; kwargs...)
+                end
+            end
+        end
+    end
+
+    # for now consider Alder (no-leaf) and Avery (leaf)
+    new_ACArc = PSY.Arc(;
+        from = PSY.get_component(PSY.ACBus, main_sys, "Alder"),
+        to = PSY.get_component(PSY.ACBus, main_sys, "Alder_twin"),
+    )
+    PSY.add_component!(main_sys, new_ACArc)
+
+    if HVDC_line
+        new_HVDCLine = PSY.TwoTerminalHVDCLine(;
+            name = "HVDC_interconnection",
+            available = true,
+            active_power_flow = 0.0,
+            arc = get_component(Arc, main_sys, "Alder -> Alder_twin"),
+            active_power_limits_from = (min = -1000.0, max = 1000.0),
+            active_power_limits_to = (min = -1000.0, max = 1000.0),
+            reactive_power_limits_from = (min = -1000.0, max = 1000.0),
+            reactive_power_limits_to = (min = -1000.0, max = 1000.0),
+            loss = (l0 = 0.0, l1 = 0.1),
+            services = Vector{Service}[],
+            ext = Dict{String, Any}(),
+        )
+        PSY.add_component!(main_sys, new_HVDCLine)
+    else
+        new_ACLine = PSY.MonitoredLine(;
+            name = "AC_interconnection",
+            available = true,
+            active_power_flow = 0.0,
+            reactive_power_flow = 0.0,
+            arc = get_component(Arc, main_sys, "Alder -> Alder_twin"),
+            r = 0.042,
+            x = 0.161,
+            b = (from = 0.022, to = 0.022),
+            rate = 1.75,
+            # For now, not binding
+            flow_limits = (from_to = 2.0, to_from = 2.0),
+            angle_limits = (min = -1.57079, max = 1.57079),
+            services = Vector{Service}[],
+            ext = Dict{String, Any}(),
+        )
+        PSY.add_component!(main_sys, new_ACLine)
+    end
+
+    for bat in get_components(GenericBattery, main_sys)
+        set_base_power!(bat, get_base_power(bat) * 10)
+    end
+
+    for r in get_components(
+        x -> get_prime_mover_type(x) == PrimeMovers.CP,
+        RenewableDispatch,
+        main_sys,
+    )
+        clear_services!(r)
+        remove_component!(main_sys, r)
+    end
+
+    for dev in get_components(RenewableFix, main_sys)
+        clear_services!(dev)
+    end
+
+    for dev in
+        get_components(x -> get_fuel(x) == ThermalFuels.NUCLEAR, ThermalStandard, main_sys)
+        clear_services!(dev)
+    end
+
+    for dev in get_components(HydroGen, main_sys)
+        clear_services!(dev)
+    end
+
+    bus_to_change = PSY.get_component(ACBus, main_sys, "Arne_twin")
+    PSY.set_bustype!(bus_to_change, PSY.ACBusTypes.PV)
+
+    # cost perturbation must be the same for each sub-system
+    rand_ix = 1
+    for g in get_components(
+        x -> x.prime_mover_type in [PrimeMovers.CT, PrimeMovers.CC],
+        ThermalStandard,
+        main_sys,
+    )
+        old_pwl_array = get_variable(get_operation_cost(g)) |> get_cost
+        new_pwl_array = similar(old_pwl_array)
+        for (ix, tup) in enumerate(old_pwl_array)
```

Comment:
I think
```suggestion
        for (ix, (y, x)) in enumerate(old_pwl_array)
```
and corresponding changes below would be clearer than `tup[1]` and `tup[2]`